File: producer_yaml/src/main/java/com/example/intoxication/NoOpResponseProvider.java
package com.example.intoxication;

import org.springframework.stereotype.Service;

/**
 * @author Marcin Grzejszczak
 */
@Service
class NoOpResponseProvider implements ResponseProvider {
	@Override public Response thereYouGo(Customer personToCheck) {
		return null;
	}
}

File: producer_kotlin/src/main/kotlin/com/example/intoxication/NoOpResponseProvider.kt
package com.example.intoxication

import org.springframework.stereotype.Service

/**
 * @author Marcin Grzejszczak
 */
@Service
internal class NoOpResponseProvider : ResponseProvider {
    override fun thereYouGo(personToCheck: Customer): Response {
        return Response(DrunkLevel.SOBER, DrunkLevel.SOBER)
    }
}
